### PR TITLE
[kubeadm reference] Fix Config API's value syntax for external etcd cluster and `kubeletExtraArgs`

### PR DESCRIPTION
**Pull request by g1eng**
Hello.

A configuration example on [Config API reference](https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/) to bootstrap cluster using `kubeadm init`, contains several invalid syntax on values for latest k8s.

### 1. Config API for external etcd cluster must contain endpoints within `proto://addr:port` style, not `addr:port` style.

This is correctly (but implicitly) mentioned in [another page](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/high-availability/#external-etcd-nodes)

If you try `kubeadm init` with the current example on the documentation to try external etcd architecture, it results a runtime error as following:

```
[etcd.external.endpoints: Invalid value: "10.101.0.11:2380": URL parse error: parse "10.101.0.11:2380": first path segment in URL cannot contain colon, etcd.external.endpoints: Invalid value: "10.101.0.21:2380": URL parse error: parse "10.101.0.21:2380": first path segment in URL cannot contain colon, etcd.external.endpoints: Invalid value: "10.101.0.51:2380": URL parse error: parse "10.101.0.51:2380": first path segment in URL cannot contain colon]
To see the stack trace of this error execute with --v=5 or higher
```

### 2. kubeletExtraArgs must be an array of `quoted strings`

See [this](https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/types.go#L230).

If you run `kubeadm init` with configuration example part from [Config API reference](https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/), it fails.

### Additional information

Also [the documentation of Config API v1beta4](https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta4/) contains same (maybe invalid) syntax, but I have no idea about how to try v1beta4 or whether I should examine compatibility for past kubernetes releases possibly affected by this PR or not.

(So the fixation for v1beta4 documentation is not included in the PR, at now)

If anyone has knowledge about the situation, give me suggestion how should we fix the documentation error.

Thanks.

### Appendix for reproduction

This issue is reproducible under kubenetes v1.28.5.

#### an example of invalid configuration

```
apiVersion: kubeadm.k8s.io/v1beta3
kind: InitConfiguration
nodeRegistration:
  name: somehost.example.com
  kubeletExtraArgs:
    v: 5
localAPIEndpoint:
  advertiseAddress: 0.0.0.0
  bindPort: 6443
---
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
etcd:
  external:
    endpoints:
    - "10.10.108.11:2380"
    - "10.10.108.12:2380"
    - "10.10.108.13:2380"
kubernetesVersion: 1.28.5
networking:
  podSubnet: 10.140.0.0/16
```

#### an example of valid configuration

```
apiVersion: kubeadm.k8s.io/v1beta3
kind: InitConfiguration
nodeRegistration:
  name: somehost.example.com
  kubeletExtraArgs:
    v: "5"
localAPIEndpoint:
  advertiseAddress: 0.0.0.0
  bindPort: 6443
---
apiVersion: kubeadm.k8s.io/v1beta3
kind: ClusterConfiguration
etcd:
  external:
    endpoints:
    - "http://10.10.108.11:2380"
    - "http://10.10.108.12:2380"
    - "http://10.10.108.13:2380"
kubernetesVersion: 1.28.5

networking:
  podSubnet: 10.140.0.0/16
```

### CODE REVIEW

**Comment by T-Lakshmi** [comment]
@g1eng,
Thank you for your PR contribution in kubernetes community.

Please sign CLA to proceed with review process.
https://github.com/kubernetes/community/blob/master/CLA.md

**Comment by g1eng** [comment]
@T-Lakshmi I see. wait a moment...

**Review by dipesh-rawat — COMMENTED** [review]
This [kubeadm Configuration (v1beta3)](https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta3/#auto-generated-edit-notice) reference page is auto-generated from the upstream source code, any fixes would need to be addressed there. Please refer to the upstream contribution guidelines outlined at: https://kubernetes.io/docs/contribute/generate-ref-docs/contribute-upstream/, https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/

**Comment by g1eng** [comment]
Thanks for your guidance.
Okay, I'll try to commit the upstream code which originates this documentation bug.

By the way, how will this repository synchronize the upstream documentation generated? (I cannot find the CI/CD pipeline to automate the documentation code synchronization...)

File: content/en/docs/reference/config-api/kubeadm-config.v1beta3.md [hunk 1]
```diff
@@ -183,8 +183,8 @@ configuration types to be used during a <code>kubeadm init</code> run.</p>
 </span><span style="color:#bbb">      </span>- <span style="color:#d14">&#34;10.100.0.1&#34;</span><span style="color:#bbb">
 </span><span style="color:#bbb">  </span><span style="color:#998;font-style:italic"># external:</span><span style="color:#bbb">
 </span><span style="color:#bbb">    </span><span style="color:#998;font-style:italic"># endpoints:</span><span style="color:#bbb">
-</span><span style="color:#bbb">    </span><span style="color:#998;font-style:italic"># - &#34;10.100.0.1:2379&#34;</span><span style="color:#bbb">
-</span><span style="color:#bbb">    </span><span style="color:#998;font-style:italic"># - &#34;10.100.0.2:2379&#34;</span><span style="color:#bbb">
+</span><span style="color:#bbb">    </span><span style="color:#998;font-style:italic"># - &#34;https://10.100.0.1:2379&#34;</span><span style="color:#bbb">
+</span><span style="color:#bbb">    </span><span style="color:#998;font-style:italic"># - &#34;https://10.100.0.2:2379&#34;</span><span style="color:#bbb">
```
Comment (neolit123) [inline content/en/docs/reference/config-api/kubeadm-config.v1beta3.md:187]: including https:// as example is correct. the validation will fail non HTTPs scheme URls
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/validation/validation.go#L386

File: content/en/docs/reference/config-api/kubeadm-config.v1beta3.md [hunk 2]
```diff
@@ -156,7 +156,7 @@ configuration types to be used during a <code>kubeadm init</code> run.</p>
 </span><span style="color:#bbb">      </span><span style="color:#000;font-weight:bold">value</span>:<span style="color:#bbb"> </span><span style="color:#d14">&#34;someValue&#34;</span><span style="color:#bbb">
 </span><span style="color:#bbb">      </span><span style="color:#000;font-weight:bold">effect</span>:<span style="color:#bbb"> </span><span style="color:#d14">&#34;NoSchedule&#34;</span><span style="color:#bbb">
 </span><span style="color:#bbb">  </span><span style="color:#000;font-weight:bold">kubeletExtraArgs</span>:<span style="color:#bbb">
-</span><span style="color:#bbb">    </span><span style="color:#000;font-weight:bold">v</span>:<span style="color:#bbb"> </span><span style="color:#099">4</span><span style="color:#bbb">
+</span><span style="color:#bbb">    </span><span style="color:#000;font-weight:bold">v</span>:<span style="color:#bbb"> </span><span style="color:#099">&#34;4&#34;</span><span style="color:#bbb">
```
Comment (neolit123) [inline content/en/docs/reference/config-api/kubeadm-config.v1beta3.md:159]: is it actually a problem if the 5 is not quoted?

**Review by neolit123 — COMMENTED** [review]
the changes you are suggesting seem OK but this is not the right place to fix them.

this is generated HTML code from here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/doc.go

for v1beta4 here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta4/doc.go

sending PR for both APIs is OK, but this repo is not the right place.

also please sign the CLA or your PRs will not be accepted.

/close

**Comment by tengqm** [comment]
Upstream source: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/apis/kubeadm/v1beta3/doc.go#L210-L212

**Comment by g1eng** [comment]
Thanks!

**Pull request closed without merging**